jupyter_collaboration/loaders.py

```diff
@@ -0,0 +1,94 @@
+import asyncio
+from logging import Logger, getLogger
+from typing import Any, Callable, Coroutine, Dict, Optional
+
+from jupyter_server.utils import ensure_async
+
+
+class FileLoader:
+    def __init__(
+        self,
+        path: str,
+        file_format: str,
+        file_type: str,
+        contents_manager: Any,
+        log: Optional[Logger],
+        poll_interval: Optional[float] = None,
+    ) -> None:
+        self._path: str = path
+        self._file_format: str = file_format
+        self._file_type: str = file_type
+        self._last_modified = None
+
+        self._lock = asyncio.Lock()
+        self._poll_interval = poll_interval
+        self._contents_manager = contents_manager
+
+        self._log = log or getLogger(__name__)
+        self._subscriptions: Dict[str, Callable[[str], Coroutine[Any, Any, None]]] = {}
+
+        if self._poll_interval:
+            self._watcher = asyncio.create_task(self.watch_file())
+
+    @property
+    def path(self):
+        return self._path
+
+    def clean(self) -> None:
+        self._watcher.cancel()
+
+    def rename_file(self, path: str) -> None:
+        self._path = path
+
+    def number_of_subscriptions(self) -> int:
+        return len(self._subscriptions)
+
+    def observe(self, id: str, callback: Callable[[str], Coroutine[Any, Any, None]]) -> None:
+        self._subscriptions[id] = callback
+
+    def unobserve(self, id: str) -> None:
+        del self._subscriptions[id]
+
+    async def load_content(self, format: str, file_type: str, content: bool) -> Dict[str, Any]:
+        return await ensure_async(
```

Comment:
Don't we want to use the lock here?

Reply:
I think it's problematic to load a file while it is being saved. You are probably missing a lock [here](https://github.com/jupyterlab/jupyter_collaboration/pull/121/files#diff-54b0d5450e8be93702b7ef60bc9596a423a203f5591cb9b61c6ff29dbcadf39bR113).

Reply:
We are using the contents manager. I'm assuming it is thread-safe. Otherwise, we would have the same issue with the contents API.